python/adaptivity.py.in

```diff
@@ -333,7 +333,65 @@ def detect_colinearity(mesh, tol = 30):
 
     return bfaces_func
 
-
+def adapt_boundary_regions(original_mesh, adapted_mesh, original_regions):
```

Comment:
I'm not sure why this function is required. Pragmatic does preserve boundary tags that are passed through pragmatic_set_boundary and returned by pragmatic_get_boundaryTags. The former is called in the adapt(...) routine, you can just recover them with the latter. 

The way it works is the boundary tags array is "aligned" with the cells array, and for each vertex you get the tag of the opposite facet in the element. So tags[3*i+k] gives you the tag of the facet opposite to the k-th vertex of the i-th triangle